Mathlib/GroupTheory/SpecificGroups/PresentedDihedral.lean

```diff
@@ -0,0 +1,231 @@
+/-
+Copyright (c) 2023 Newell Jensen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Newell Jensen
+-/
+import Mathlib.GroupTheory.PresentedGroup
+import Mathlib.GroupTheory.SpecificGroups.Dihedral
+
+/-
+# Presentation of Dihedral Groups
+
+This file defines a group presentation of the dihedral groups, `PresentedDihedralGroup n`,
+using the presentation `⟨a, b | a² = 1, b² = 1, (a * b)ⁿ = 1⟩`.
+
+## Main definitions
+
+* TODO
+
+## Main results
+
+* `PresentedDihedralGroup n ≃* DihedralGroup n` proves that `PresentedDihedralGroup n` is
+  isomorphic to `DihedralGroup n`.
+
+## References
+
+* https://en.wikipedia.org/wiki/Presentation_of_a_group
+
+## Tags
+presentation group, dihedral group
+-/
+
+
+variable {n : ℕ}
+
+inductive Generator (n : ℕ)
```

Comment:
Are you referring to the inductive definition?  I was following what was already in Mathlib since the `DihedralGroup` definition is in the root namespace as well.